File: week15/week15-3.cpp
//week15-3.cpp
//LeetCode1208.
int equalSubstring(char* s, char* t, int maxCost) {
    int N=strlen(s);//先知道字串的長度
    int ans =0,j=0;
    for(int i=0;i<N;i++){//字串的迴圈
        maxCost-=abs(s[i]-t[i]);
        while(maxCost<0){
            maxCost += abs(s[j]-t[j]);
            j+=1;
        }
        if(i-j+1>ans) ans=i-j+1;//更新答案
    }
    return ans;
}

File: week15/week15-3.py
#week15-3.py
#LeetCode1208.
class Solution:
    def equalSubstring(self, s: str, t: str, maxCost: int) -> int:
        #先用python 把字串迴圈寫出來
        N=len(s) #先知道字串的長度
        ans=0
        j=0
        for i in range(N): #字串的迴圈
            maxCost -=abs(ord(s[i] )-ord(t[i]))
            while maxCost<0:
                maxCost +=abs(ord(s[j] )-ord(t[j]))
                j   += 1
            ans=max(ans,i-j+1)
            #print(s[i],t[i]) #字串的陣列
        return ans #課堂作業1，先隨便給個0當答案